internal/tui/model.go

```diff
@@ -2,146 +2,150 @@ package tui
 
 import (
 	"envctl/internal/utils"
-	"fmt"
+	"fmt" // Import os for stderr
 	"strings"
 	"time"
 
 	"github.com/charmbracelet/bubbles/textinput"
+	"github.com/charmbracelet/bubbles/viewport"
 	tea "github.com/charmbracelet/bubbletea"
 	"github.com/charmbracelet/lipgloss"
 )
 
-// Define an enum for the current step in the new connection input process.
+// newInputStep defines the different stages of the new connection input process.
+// This helps manage the UI flow when a user initiates a new connection.
 type newInputStep int
 
 const (
-	mcInputStep newInputStep = iota
-	wcInputStep
-	maxCombinedOutputLines = 200 // Define the constant for log trimming
+	mcInputStep newInputStep = iota // Represents the stage where the user inputs the Management Cluster name.
+	wcInputStep                     // Represents the stage where the user inputs the Workload Cluster name.
+
+	// maxCombinedOutputLines defines the maximum number of lines to keep in the combinedOutput log.
```

Comment:
The model struct has grown quite large. Consider breaking it down into smaller, focused sub-structures for better maintainability:

```go
type clusterState struct {
    name       string
    health     clusterHealthInfo
    kubeContext string
}

type portForwardState struct {
    processes     map[string]*portForwardProcess
    order         []string
    focusedPanelKey string
}

type uiState struct {
    width, height int
    debugMode     bool
    colorMode     string
    helpVisible   bool
    // ... other UI related fields
}

type model struct {
    managementCluster clusterState
    workloadCluster   clusterState
    portForwards     portForwardState
    ui              uiState
    // ... other fields
}
```